pkg/storage/db.go

```diff
@@ -151,10 +152,35 @@ func (db *DB) Init(ctx context.Context, postgresConfig *PostgresConfig) error {
 
 		}
 		initerr = db.RunScript(ctx, createScript)
+
+		if initerr == nil {
+			db.startPartitionGeneratorTimer(ctx)
+		}
 	})
 	return initerr
 }
 
+func (db *DB) startPartitionGeneratorTimer(ctx context.Context) {
+	// run every 12 hours, with the assumption that partition tables will never be
+	// this granular, thus partition tables are created well in advance of their need
+	ticker := time.NewTicker(12 * time.Hour)
+	quit := make(chan struct{})
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				if err := db.generatePartition(ctx); err != nil {
+					logger := domain.LoggerFromContext(ctx)
+					logger.Error(logs.StorageError{Reason: err.Error()})
+				}
+			case <-quit:
```

Comment:
It doesn't look like this channel is ever written to. Ideally, we would be able to hook into some sort of lifecycle event to start/stop this, where the start function would execute this code path and the stop function would write to the quit channel causing the timer to stop.

The scenario I am thinking of is bringing up a new stack, while the old stack still lingers for a while in case of a rollback. In that case, the old stack will still be running and could possible trigger an extra run of this timer. Ultimately, since the table is locked, it is benign. However, as it is currently written, I don't think this code path will ever be executed.